libraries/AC_Fence/AC_Fence.cpp

```diff
@@ -276,11 +336,10 @@ bool AC_Fence::check_destination_within_fence(const Location_Class& loc)
     // polygon fence check
     if ((get_enabled_fences() & AC_FENCE_TYPE_POLYGON) && _boundary_num_points > 0) {
         // check ekf has a good location
-        Location temp_loc;
-        if (_inav.get_location(temp_loc)) {
-            const struct Location &ekf_origin = _inav.get_origin();
-            Vector2f position = location_diff(ekf_origin, loc) * 100.0f;
-            if (_poly_loader.boundary_breached(position, _boundary_num_points, _boundary, true)) {
+        Vector2f posNE;
+        if (_ahrs.get_relative_position_NE_origin(posNE)) {
```

Comment:
Copied from #7345:

> The previous code was a bit strange so I see why you got this wrong. See that the `get_location` call is with a `temp_loc` variable that is never used later. You need to get the origin (and check it is valid) and then do the `location_diff` and use that Vector for checking the boundary.

Reply:
Confused.

On load we translate the boundary to origin-relative coordinates.

On check we check using our origin-relative position.

This does work in testing - which it probably couldn't if this was being screwed up?

Reply:
Please look at the code again. This code is in a method with the following signature `bool check_destination_within_fence(const Location_Class& loc)`. So where is this code checking that `loc` is inside the fence? What this new code is checking is that the current position is inside the fence - that's not what this method is for!

Reply:
OK, now I understand.  PR coming.

Reply:
https://github.com/ArduPilot/ardupilot/pull/7410